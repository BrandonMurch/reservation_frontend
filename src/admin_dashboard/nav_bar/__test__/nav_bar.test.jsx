// Dependencies
import React from 'react';
import { unmountComponentAtNode } from 'react-dom';
import { create } from 'react-test-renderer';
import { renderWithRouter } from 'test_utils';

// Components
import NavBar from '../index';

describe('<NavBar />', () => {
  let component;
  let container = null;

  beforeEach(() => {
    container = document.createElement('div');
    document.body.appendChild(container);
    component = renderWithRouter(
      <NavBar>
        <div data-testid="testChild" />
      </NavBar>,
      { route: '/admin' },
    );
  });

  afterEach(() => {
    unmountComponentAtNode(container);
    container.remove();
    container = null;
  });

  it('should match snapshot', () => {
    const tree = create(
      <NavBar>
        <div data-testid="testChild" />
      </NavBar>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render children', () => {
    const child = component.getByTestId('testChild');
    expect(child).toBeInTheDocument();
  });
});
