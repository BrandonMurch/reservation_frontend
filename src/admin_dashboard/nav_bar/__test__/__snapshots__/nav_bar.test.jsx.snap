// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBar /> should match snapshot 1`] = `
<div
  className="container"
>
  <div
    data-testid="testChild"
  />
</div>
`;
